app-typescript/components/ToggleBox/index.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import {SimpleToggleBox} from './SimpleToggleBox';
+import {CustomHeaderToggleBox} from './CustomHeaderToggleBox';
+
+export interface IPropsWithLabel {
+    variant: 'simple';
+    title: string;
+    badge?: JSX.Element;
+    children: React.ReactNode;
+    hideUsingCSS?: boolean;
+    initiallyOpen?: boolean; // defaults to false
+    className?: string;
+    margin?: 'none' | 'small' | 'normal' | 'large';
+    onOpen?(): void;
+    onClose?(): void;
+}
+
+export interface IPropsWithoutLabel {
+    variant: 'custom-header';
+    // always visible
+    header: React.ReactNode;
+    children: React.ReactNode;
+    toggleButton: string;
```

Comment:
call this `toggleButtonLabel`. It's not the entire button that's passed as prop.